### ISSUE TITLE
Invalid pkg.list_upgrades parsing on Fedora 28 Workstation

### ISSUE THREAD

**Issue by e-user** [saltstack/salt#49328]
### Description of Issue/Question
Similar to issue https://github.com/saltstack/salt/issues/47717 (which was fixed), running `salt-call --local pkg.list_upgrades` on Fedora 28 Workstation produces the following broken result in my current setup:
```
local:
    ----------
    0.076-1.fc28:
        updates
    0.1.56-2.fc28:
        updates
    0.3.2-2.fc28:
        updates
[...]
    Available:
        Upgrades
```

The bare output from the command run by Salt, `dnf --quiet list upgrades`, looks like this:
```
Available Upgrades        
armadillo.x86_64                           8.600.1-1.fc28                 updates       
authselect.x86_64                          1.0-1.fc28                     updates       
authselect-compat.x86_64                   1.0-1.fc28                     updates
[...]
```

### Setup
This is being run from a masterless setup, no deviation from the standard configuration provided by the upstream package. Salt was installed from the Fedora upstream repositories.

### Steps to Reproduce Issue
`salt-call --local pkg.list_upgrades`

### Versions Report
```
Salt Version:
           Salt: 2018.3.2
 
Dependency Versions:
           cffi: 1.11.5
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.10
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.15 (default, May 16 2018, 17:50:09)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 17.0.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 5.0.2
            ZMQ: 4.1.6
 
System Versions:
           dist: fedora 28 Twenty Eight
         locale: UTF-8
        machine: x86_64
        release: 4.17.14-202.fc28.x86_64
         system: Linux
        version: Fedora 28 Twenty Eight
```

**Comment by e-user** [comment]
@garethgreenaway Good question. Should be easy to set up a container with F27 and test it. I can try if I get around to it, if it helps!

**Comment by e-user** [comment]
@garethgreenaway I just checked in a fresh `fedora:27` Docker container and the output as well as the bug are the same with Salt 2018.3.2.

```
local:
    ----------
    10.31-10.fc27:
        updates
    7.55.1-13.fc27:
        updates
    8.42-3.fc27:
        updates
    Available:
        Upgrades
```

**Comment by garethgreenaway** [comment]
@e-user Apologies for the delay in circling back to this.  I can confirm this happening with Fedora and Salt version 2018.3.2, but it appears to be fixed on the HEAD of 2018.3 so will be fixed in 2018.3.3 once that is released.
